src/ConnectedMode/Suppressions/SuppressionIssueStoreUpdater.cs

```diff
@@ -30,16 +37,121 @@ internal interface ISuppressionIssueStoreUpdater
+    {
+        private readonly ISonarQubeService server;
+        private readonly IServerQueryInfoProvider serverQueryInfoProvider;
+        private readonly IServerIssueStoreWriter storeWriter;
+        private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
+
+        private CancellationTokenSource updateAllCancellationTokenSource = new CancellationTokenSource();
+
+        [ImportingConstructor]
+        public SuppressionIssueStoreUpdater(ISonarQubeService server,
+            IServerQueryInfoProvider serverQueryInfoProvider,
+            IServerIssueStoreWriter storeWriter,
+            ILogger logger)
+            : this(server, serverQueryInfoProvider, storeWriter, logger, ThreadHandling.Instance)
+        {
+        }
+
+        internal /* for testing */ SuppressionIssueStoreUpdater(ISonarQubeService server,
+            IServerQueryInfoProvider serverQueryInfoProvider,
+            IServerIssueStoreWriter storeWriter,
+            ILogger logger,
+            IThreadHandling threadHandling)
+        {
+            this.server = server;
+            this.serverQueryInfoProvider = serverQueryInfoProvider;
+
+            this.storeWriter = storeWriter;
+            this.logger = logger;
+            this.threadHandling = threadHandling;
+        }
+
+        #region ISuppressionIssueStoreUpdater
+
+        public async Task UpdateAllServerSuppressionsAsync()
+        {
+            await threadHandling.SwitchToBackgroundThread();
+
+            try
+            {
+                logger.WriteLine(Resources.Suppressions_Fetch_AllIssues);
+
+                CancelCurrentOperation();
+                var localCopyOfCts = updateAllCancellationTokenSource = new CancellationTokenSource();
+                
+                (string projectKey, string serverBranch) queryInfo = await serverQueryInfoProvider.GetProjectKeyAndBranchAsync(localCopyOfCts.Token);
+
+                localCopyOfCts.Token.ThrowIfCancellationRequested();
+
+                if (queryInfo.projectKey == null || queryInfo.serverBranch == null)
+                {
+                    return;
+                }
+
+                localCopyOfCts.Token.ThrowIfCancellationRequested();
+
+                var allSuppressedIssues = await server.GetSuppressedIssuesAsync(queryInfo.projectKey, queryInfo.serverBranch, localCopyOfCts.Token);
+                storeWriter.AddIssues(allSuppressedIssues, clearAllExistingIssues: true);
+
+                logger.WriteLine(Resources.Suppression_Fetch_AllIssues_Finished);
+            }
+            catch (OperationCanceledException)
+            {
+                logger.WriteLine(Resources.Suppressions_FetchOperationCancelled);
+            }
+            catch(Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                logger.LogVerbose(Resources.Suppression_FetchError_Verbose, ex);
+                logger.WriteLine(Resources.Suppressions_FetchError_Short, ex.Message);
```

Comment:
I think I've done this in a few places more recently - log a short error message that will appear at normal verbosity, and a second verbose message with all of the detail to help with support/debugging.

Reply:
Yea I think that makes sense